packages/pyright-internal/src/languageService/autoImporter.ts

```diff
@@ -577,9 +577,9 @@ export class AutoImporter {
     }
 
     protected shouldIncludeVariable(autoSymbol: AutoImportSymbol, isStub: boolean) {
-        // If it is not a stub file and symbol is Variable, we only include it if
+        // If it is not a stub file, is not in __all__ and symbol is Variable, we only include it if
         // name is public constant or type alias
-        if (isStub || autoSymbol.kind !== SymbolKind.Variable) {
+        if (isStub || autoSymbol.kind !== SymbolKind.Variable || autoSymbol.inDunderAll) {
```

Comment:
thanks for the contribution. i think we should just always include the import suggestion to address #1007

Reply:
Good point, I've pushed a change for this and updated the PR, what do you think?

Reply:
looks good, thanks! i also want to maybe add a new diagnostic rule for importing mutable variables like i suggested in #1007, i'll try to look into that within the next few days

Reply:
on second thought i think such a rule would cause way too many false positives so i think what you have here is good as-is